docs/DeveloperGuide.md

```diff
@@ -8,22 +8,72 @@
 
 {Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
 
+### Update patient information
+Step 1:Patient inform the nurse of their change in personal information. Nurse will open up the Duke system and request to submit the change. Duke will call the Parser#etPatientInfo() to update the patient information. 
+
+Step 2:
+The parser will call the database storing patient information to update the change.
+
+The flow is presented in the sequence diagram (will be included later after pull request is permitted):
+
+
+### Exit program
+Step 1: User type "bye" as input to exit the program. IHospital will call Parser#getCommand()
+to return the user command received.
+
+Step 2: After checking the user command is "bye", IHospital will
+call Ui#sayGoodbye to print Goodbye message
+
+The following sequence diagram shows how the exit operation works:
+
+![](ByeCommand.png)
+
+### Add Doctor
+Step 1: User type add command to add a doctor into the system and provide information required.
+IHospital will call Parser#getCommand() to return the user command received, it will then call
+Parser#parse to understand that User wants to add a doctor, and Parser#parse will return an 
+AddDoctorCommand object.
+
+Step 2: IHospital will call execute of the AddDoctorCommand object and AddDoctorCommand will
+create a new Doctor with the given information and add it into DoctorList using 
+DoctorList#addDoctor(doctor).
+
+The following sequence diagram shows how the add operation works:
+
+![](AddDoctor.png)
+
+### Delete Doctor
+Step 1: User types "delete" to delete a specific doctor from the system
+This will be passed into Duke. Duke will pass the comment to the parser,
+which will parse through the command to remove the doctor.
+
+Step 2: The parser will call on deleteDoctor to the doctor list.
+The doctor list acts as a database storage containing the information of a list of doctors.
+Once the doctor is removed from the doctor list, the system informs the user that the doctor is successfully removed
+
+![](DeleteDoctor.jpg)
```

Comment:
Consider using the same software as the previous 2 diagrams, I understand that NTU provides visual paradigm, but it would be good to standardize the diagrams across the DG.

From what I know the numbering is auto added when using visual paradigm, so consider removing it? If I am not wrong, its not wrong to have numbering, but you may have to check with Prof if the numberings are allowed.

![image](https://user-images.githubusercontent.com/15943697/160964594-1c2b1e91-daf6-4aaf-9796-36a867b90f50.png)
The alignment for 1.1.1:deleteDoctor() is a little off, consider moving it higher.

![image](https://user-images.githubusercontent.com/15943697/160964784-fcdc042d-e096-476a-9f29-c551363a7f82.png)
The return arrows from methods should be dotted lines instead of solid lines.

You may want to check with Prof if there is a need for an activation bar for the User.